baybe/serialization/mixin.py

```diff
@@ -0,0 +1,52 @@
+"""Serialization mixin class."""
+
+import json
+from typing import Type, TypeVar
+
+from baybe.serialization.core import converter
+
+_T = TypeVar("_T")
+
+
+class SerialMixin:
```

Comment:
[Putting this here since there is no better place]
**TL;DR:** Change line 
<img width="534" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/d7150f32-2c23-43c6-8992-f0badfafc100">
in `conf.py` to 
<img width="540" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/eb3bfc40-b747-46cb-bcd3-be4b8df53d53">

to fix doc building issue. Tested locally, works.

**Explanation:** `sphinx` tries to retrieve the init function of the classes for the autodocumentation. However, we have some rare instances where we do not use `attrs` for automatically creating them and do not provide them ourselves. For this reason, we need to explicitly tell sphinx "Please ignore the following functions" which is done by the `nitpck_ignore_regex` list in `conf.py`. Since the structure of the class changed, we need to change the part here as well.

This issue with outdated links happen in some more places (which is only visible after this fix). I'll create two commits to fix the corresponding issues.

Reply:
Corresponding commits are pushed, so feel free to resolve if docs now works and this explanation makes sense.

Reply:
Nice, thanks for the fix 🏅 
Will rebase the branch on main to resolve the current merge conflicts and then we should see the pipeline pass!